src/platform_impl/android/mod.rs

```diff
@@ -469,6 +469,35 @@ impl<T: 'static> EventLoop<T> {
                         }
                     }
                 }
+                InputEvent::TextEvent(ime_state) => {
+                    let events = [
+                        // Send a preedit event so the application knows to expect a commit event
+                        event::Ime::Preedit("".to_string(), None),
+                        // Delete all of the current text
+                        event::Ime::DeleteSurroundingText {
+                            before_length: usize::MAX,
+                            after_length: usize::MAX,
+                        },
```

Comment:
Deleting surrounding text with usize::MAX might be a foot gun at some point.

IME edits are relative so the current cursor because conceptually they could be used within a large body of text (e.g. think 10k word Google Doc) so I'm a bit concerned that asking the toolkit to delete like this could potentially result in it deleting someone's larger document unless the toolkit additionally tracks the bounds of the surrounding context it has give Winit and treats that like a sandbox (but that's not currently clearly specified afik).

Maybe we can reliably track the size of the current text so we can more precisely delete just the text from the last update?

Reply:
I agree with that, doing delete all is sort of weird. Given that it's all about the feedback loop with `setSurroundingText` request, maybe we can base everything on it, since
IME shouldn't touch anything outside of the user set surrounding text, no?